ovs/extensions/services/fleetctl.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2015 iNuron NV
+#
+# Licensed under the Open vStorage Modified Apache License (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.openvstorage.org/license
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Fleetctl module
+"""
+import time
+
+try:
+    import fleet.v1 as fleet
+    FLEET_CLIENT = fleet.Client('http+unix://%2Fvar%2Frun%2Ffleet.sock')
+except ImportError:
+    raise ImportError('Fleet python client is not installed. Please check documentation on how to install it')
+
+from ovs.log.logHandler import LogHandler
+from ovs.extensions.services.systemd import Systemd
+
+logger = LogHandler.get('extensions', name='fleetctl')
+
+
+class FleetCtl(object):
+    """
+    Contains all logic related to managing services through fleet
+    """
+
+    @staticmethod
+    def prepare_template(base_name, target_name, client):
+        return Systemd.prepare_template(base_name, target_name, client)
+
+    @staticmethod
+    def add_service(name, client, params=None, target_name=None, additional_dependencies=None):
+        """
+        This will generate a .service (temporary) file to feed to fleet to start a service
+         service will become "name@<client.ip>.service"
+        """
+        if params is None:
+            params = {}
+        client_ip = FleetCtl._get_client_ip(client)
+
+        if FleetCtl.has_service(name, client):
+            logger.info('Not re-adding service {0} to machine {1}'.format(name, client_ip))
+            return
+
+        name = Systemd._get_name(name, client, '/opt/OpenvStorage/config/templates/systemd/')
+        template_service = '/opt/OpenvStorage/config/templates/systemd/{0}.service'
+
+        if not client.file_exists(template_service.format(name)):
+            # Given template doesn't exist so we are probably using system
+            # init scripts
+            return
+
+        template_file = client.file_read(template_service.format(name))
+
+        for key, value in params.iteritems():
+            template_file = template_file.replace('<{0}>'.format(key), value)
+        if '<SERVICE_NAME>' in template_file:
+            service_name = name if target_name is None else target_name
+            template_file = template_file.replace('<SERVICE_NAME>', service_name.lstrip('ovs-'))
+
+        dependencies = ''
+        if additional_dependencies:
```

Comment:
if check obsolete
You should overwrite the additional dependencies with an empty list at start of function similar to what you do with params
Then you can always iterate the list, without having to check first if anything in it